sys/include/event/timeout.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_event
+ * @brief       Provides functionality to trigger events after timeout
```

Comment:
Question for clarification: Is this to replace `evtimer`?

Reply:
Nope. This is so that users of the event module can also have timed events. It would probably be a good idea to implement event_timer_t on top of evtimer. The similarity of the names is unfortunately confusing...

Reply:
Well this PR provides an event queue implementation, `evtimer` is a timed event queue. Shouldn't there be some way to merge them? What is `event_timer_t`?

Reply:
There is, maybe, but not a trivial one... Also, the use-cases are quite different. evtimer basically runs thread-independent, with it's event handlers being run in ISR context.
This event queue is much more under user control.

> What is event_timer_t?

I meant ```event_timeout_t```, the method of haveing timed events within event queues.